### PR TITLE
Broker Release Pipeline

### PR DESCRIPTION
**Pull request by p3dr0rv**
* Add broker release pipeline
* Update name of pipeline that build brokerhost (this new name has an unique identifier needed for other pipelines)
* Update documentation

### CODE REVIEW

**Review by shoatman — APPROVED** [review]

File: azure-pipelines/ui-automation/broker-test.yml [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+  azureSamplePipelineId: 1458
+  msalApp: msalautomationapp-dist-AutoBroker-debug.apk
+  msalTestApp: msalautomationapp-dist-AutoBroker-debug-androidTest.apk
+  brokerApp: brokerautomationapp-dist-AutoBroker-debug.apk
+  brokerTestApp: brokerautomationapp-dist-AutoBroker-debug-androidTest.apk
+  azureSampleApk: AzureSample-external-release.apk
+  companyPortalApk: CompanyPortal.apk
+  authenticatorApk: Authenticator.apk
+  deviceId: flame
+  deviceAndroidVersion: 29
```
Comment (shahzaibj) [inline azure-pipelines/ui-automation/broker-test.yml:34]: Could these also become parameters?

File: azure-pipelines/ui-automation/broker-test.yml [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+parameters:
+- name: companyPortalLink
+  displayName: Company Portal RC Link
+  type: string
+- name: authenticatorLink
+  displayName: Authenticator RC Link
+  type: string
+
+stages:
+  # msalautomationapp
```
Comment (shahzaibj) [inline azure-pipelines/ui-automation/broker-test.yml:47]: Does this need to be removed?

File: azure-pipelines/ui-automation/broker-test.yml [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+- stage: 'msalautomationapp'
+  dependsOn: []    # this removes the implicit dependency on previous stage and causes this to run in parallel
```
Comment (shahzaibj) [inline azure-pipelines/ui-automation/broker-test.yml:49]: Which previous stage?

File: azure-pipelines/ui-automation/broker-test.yml [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+  displayName: MSAL Automation APKs
```
Comment (shahzaibj) [inline azure-pipelines/ui-automation/broker-test.yml:50]: What about those APKs? Build them?

File: azure-pipelines/ui-automation/broker-test.yml [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+variables:
+  engineeringProjectId: 'fac9d424-53d2-45c0-91b5-ef6ba7a6bf26'
+  azureSamplePipelineId: 1458
+  msalApp: msalautomationapp-dist-AutoBroker-debug.apk
+  msalTestApp: msalautomationapp-dist-AutoBroker-debug-androidTest.apk
+  brokerApp: brokerautomationapp-dist-AutoBroker-debug.apk
+  brokerTestApp: brokerautomationapp-dist-AutoBroker-debug-androidTest.apk
+  azureSampleApk: AzureSample-external-release.apk
+  companyPortalApk: CompanyPortal.apk
+  authenticatorApk: Authenticator.apk
```
Comment (shahzaibj) [inline azure-pipelines/ui-automation/broker-test.yml:32]: Aren't these supplied by URL?

File: docs/Automation/release_automation_scripts.md [hunk 6]
```diff
@@ -57,6 +61,18 @@ The script takes two arguments as follows:
 - **authenticatorApkPath** - the path to Authenticator APK pointing to broker version (typically an RC build) that we are trying to test
 - **companyPortalApkPath** - the path to Company Portal APK pointing to broker version (typically an RC build) that we are trying to test
 
+### Pipelines
+
+- Run the [Azure Sample PROD App Pipeline](https://dev.azure.com/IdentityDivision/IDDP/_build?definitionId=1278) by supplying the `distMsalVersion` in the pipeline variables
```
Comment (shahzaibj) [inline docs/Automation/release_automation_scripts.md:66]:
```suggestion
- Run the [Azure Sample PROD App Pipeline](https://dev.azure.com/IdentityDivision/IDDP/_build?definitionId=1278) by supplying the `distMsalVersion` in the pipeline variables.
```

Comment (shahzaibj) [reply docs/Automation/release_automation_scripts.md:66]: Can we supply a default distMsalVersion? What happens if I leave it empty? I reckon build.gradle will have one already

File: docs/Automation/release_automation_scripts.md [hunk 7]
```diff
@@ -57,6 +61,18 @@ The script takes two arguments as follows:
...
+
+- Wait for the above pipeline to finish, then move on to next step
```
Comment (shahzaibj) [inline docs/Automation/release_automation_scripts.md:68]:
```suggestion
- Wait for the above pipeline to finish, then move on to next step.
```

File: docs/Automation/release_automation_scripts.md [hunk 8]
```diff
@@ -57,6 +61,18 @@ The script takes two arguments as follows:
...
+
+- Run the [Broker Release Test Pipeline](https://identitydivision.visualstudio.com/Engineering/_build?definitionId=1490) and supply
+  * `msal_branch` and `broker_branch` in the pipeline variables
+  * [Company Portal RC Link](https://install.appcenter.ms/orgs/Intune-Windroid/apps/Intune-Android-Company-Portal-Broker) and [Authenticator RC Link](https://install.appcenter.ms/orgs/authapp-t7qc/apps/Microsoft-Authenticator-Android-Broker) in the pipeline arguments
```
Comment (shahzaibj) [inline docs/Automation/release_automation_scripts.md:72]:
```suggestion
  * [Company Portal RC Link](https://install.appcenter.ms/orgs/Intune-Windroid/apps/Intune-Android-Company-Portal-Broker) and [Authenticator RC Link](https://install.appcenter.ms/orgs/authapp-t7qc/apps/Microsoft-Authenticator-Android-Broker) in the pipeline arguments.
```

**Review by shahzaibj — APPROVED** [review]
Minor comments. LGTM!

Thanks @p3dr0rv!

**Comment by p3dr0rv** [comment]
Waiting on @iamgusain changes to address comments and merge

File: docs/Automation/release_automation_scripts.md (hunk 8 above)
Comment (p3dr0rv) [reply docs/Automation/release_automation_scripts.md]: 40133a60a4c22a1727c6a615b91bf6b54f2d9bef

File: docs/Automation/release_automation_scripts.md (hunk 7 above)
Comment (p3dr0rv) [reply docs/Automation/release_automation_scripts.md]: 40133a60a4c22a1727c6a615b91bf6b54f2d9bef

File: docs/Automation/release_automation_scripts.md (hunk 6 above)
Comment (p3dr0rv) [reply docs/Automation/release_automation_scripts.md]: 40133a60a4c22a1727c6a615b91bf6b54f2d9bef

File: azure-pipelines/ui-automation/broker-test.yml (hunk 5 above)
Comment (p3dr0rv) [reply azure-pipelines/ui-automation/broker-test.yml]: these names are defined by the universal package now

File: azure-pipelines/ui-automation/broker-test.yml (hunk 4 above)
Comment (p3dr0rv) [reply azure-pipelines/ui-automation/broker-test.yml]: 40133a60a4c22a1727c6a615b91bf6b54f2d9bef
rename here

File: azure-pipelines/ui-automation/broker-test.yml (hunk 3 above)
Comment (p3dr0rv) [reply azure-pipelines/ui-automation/broker-test.yml]: no previous step, my bad
40133a60a4c22a1727c6a615b91bf6b54f2d9bef

File: azure-pipelines/ui-automation/broker-test.yml (hunk 2 above)
Comment (p3dr0rv) [reply azure-pipelines/ui-automation/broker-test.yml]: 40133a60a4c22a1727c6a615b91bf6b54f2d9bef

File: azure-pipelines/ui-automation/broker-test.yml (hunk 1 above)
Comment (p3dr0rv) [reply azure-pipelines/ui-automation/broker-test.yml]: done 
40133a60a4c22a1727c6a615b91bf6b54f2d9bef

**Pull request merged**